openhtf/util/units/bin/units_from_xls.py

```diff
@@ -0,0 +1,160 @@
+# coding: iso-8859-1
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Read in a .xls file and generate OpenHTF Unit objects.
+
+UNECE, the United Nations Economic Commision for Europe, publishes a set of
+unit codes for international trade in the form of Excel spreadsheets (.xls
+files) here:
+
+http://www.unece.org/cefact/codesfortrade/codes_index.html
+
+This tool is used to parse those spreadsheets and turn the published standard
+code set into a .py file that OpenHTF can import.
+
+Typical usage looks like:
+
+python ./bin/units_from_xls.py --infile ./Rec20_Rev6e_2009.xls \
+--outfile ./__init__.py
+
+"""
+
+
+import argparse
+import os
+import shutil
+import sys
+from tempfile import mkstemp
+
+import xlrd
+
+
+# Column names for the columns we care about. This list must be populated in
+# the expected order: [<name label>, <code label>, <suffix label>].
+COLUMN_NAMES = ['Name',
+                'Common\nCode',
+                'Symbol']
+
+# The contents of the lines of the output file between which the generated code
+# will be inserted.
+START_DELIMITER = '######## BEGIN_UNITS ########'
+END_DELIMITER = '######## END_UNITS ########'
+
+SHEET_NAME = 'Annex I'
+REPLACEMENTS = {' ': '_',
```

Comment:
also i would name this more specifically, this isn't a global replacement dict for everything, only the `UNITS` key, maybe `unit_key_replacements` or something.